broadbean/sequence.py

```diff
@@ -478,21 +481,102 @@ def description(self):
         """
         desc = {}
 
-        for pos, elem in self._data.items():
+        for pos, elem in self._data.items():           
             desc[str(pos)] = {}
             desc[str(pos)]['channels'] = elem.description
             try:
                 sequencing = self._sequencing[pos]
-                seqdict = {'Wait trigger': sequencing[0],
-                           'Repeat': sequencing[1],
-                           'Event jump to': sequencing[2],
-                           'Go to': sequencing[3]}
+                seqdict = {'Wait trigger': sequencing['twait'],
+                           'Repeat': sequencing['nrep'],
+                           'jump_input': sequencing['jump_input'],
+                           'jump_target': sequencing['jump_target'],
+                           'Go to': sequencing['goto']}
                 desc[str(pos)]['sequencing'] = seqdict
             except KeyError:
                 desc[str(pos)]['sequencing'] = 'Not set'
-
+        desc['awgspecs'] = self._awgspecs
         return desc
 
+    def write_to_json(self, path_to_file: str) -> None:
+        """
+        Writes sequences to JSON file
+
+        Args: 
+            path_to_file: the path to the file to write to ex:
+            path_to_file/sequense.json
+        """
+        with open(path_to_file, 'w') as fp:
+            json.dump(self.description, fp, indent=4)
+
+    @classmethod
+    def init_from_json(cls, path_to_file: str):  # -> 'Sequence':
+        """
+        Reads sequense from JSON file
+
+        Args: 
+            path_to_file: the path to the file to be read ex:
+            path_to_file/sequense.json
+            This function is the inverse of write_to_json
+            The JSON file needs to be structured as if it was writen
+            by the function write_to_json
+        """
+        new_instance = cls()
+        with open(path_to_file, 'r') as fp:
+            data_loaded = json.load(fp)  
+        knowfunctions = dict([('function PulseAtoms.{}'.format(fun),
+                               getattr(PulseAtoms, fun)) for fun in
+                             dir(PulseAtoms) if '__' not in fun])
+        awgspecs = data_loaded['awgspecs']
+        SR = awgspecs['SR']
+        ElemList = list(data_loaded.keys())
+
+        for Ele in ElemList[:-1]:
+            ChannelsList = list(data_loaded[Ele]['channels'].keys())
+            elem = Element()
+            for chan in ChannelsList:
+                SegMarlist = list(data_loaded[Ele]['channels'][chan].keys())
+                Seglist = [s for s in SegMarlist if 'segment' in s]
+                bp_sum = BluePrint()
+                i = 0
+                for Seg in Seglist:
```

Comment:
I'd suggest using `for i, seg in enumerate(SegList)`

Reply:
I agree :-)